### ISSUE TITLE
Installation/Compilation fatal error C1001

### ISSUE THREAD

**Issue by MatusT** [build2/build2#45]
Hello,

I wanted to try build2 but I am getting fatal error during installation:
```
C:\Applications\build2\build2-toolchain-0.11.0\bpkg\bpkg\package-odb.cxx(2386): fatal error C1001: An internal error has occurred in the compiler.
(compiler file 'msc1.cpp', line 1519)
 To work around this problem, try simplifying or changing the program near the locations listed above.
Please choose the Technical Support command on the Visual C++
 Help menu, or open the Technical Support help file for more information
Internal Compiler Error in C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\bin\HostX64\x64\cl.exe.  You will be prompted to send an error report to Microsoft later.
INTERNAL COMPILER ERROR in 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\bin\HostX64\x64\cl.exe'
    Please choose the Technical Support command on the Visual C++
    Help menu, or open the Technical Support help file for more information
```

OS: Windows 10 1903
Visual Studio: 2019 16.2.0

Any idea how should I proceed?

**Comment by Klaim** [comment]
Same error confirmed with today's v0.12.0 stage version:
```
E:\tools\build2\build2-toolchain-0.12.0-a.0.20190725124459.aa263f4e66ed\bpkg\bpkg\package-odb.cxx(2386): fatal error C1001: An internal error has occurred in the compiler.
(compiler file 'msc1.cpp', line 1519)
...
```

@MatusT The issue is with the compiler, it's an ICE (internal compiler error). That's unfortunately common with msvc but at least they update quite fast (for a compiler).
Until they fix the issue or a workaround is implemented in Build2, you might want to use the version of the install script that use mingw instead (if you want to work on Windows).

**Comment by boris-kolpackov** [comment]
Looks like MS folks broke us again. We will look into it, thanks for the report.

In the meantime, as @Klaim suggested, you can build using the MinGW installer. You can still build your own projects with VS, you will just need to specify the C/C++ compiler explicitly (e.g., `config.cxx=cl`) since the default will be MinGW GCC.

**Comment by boris-kolpackov** [comment]
Here is the MSVC bug report: https://developercommunity.visualstudio.com/content/problem/664162/ice-on-valid-code-after-upgrading-to-1620.html

**Comment by boris-kolpackov** [comment]
Ok, seeing that the fix did not make it into MSVC 16.2.1, I've implemented a workaround so the latest staged version should work:  https://build2.org/community.xhtml#stage
